### PR TITLE
Dockerfile cleaning up

### PR DESCRIPTION
**Pull request by rolivella**
I removed some unnecessary lines.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,41 +1,39 @@
...
 ARG DEBIAN_FRONTEND=noninteractive
-
 # update and upgrade packages
 RUN apt-get -y update --fix-missing \
     && apt-get -y upgrade \
-    && apt-get -y update \
-    && apt-get -y autoremove
-
+    && apt-get -y install software-properties-common \
+    && add-apt-repository ppa:deadsnakes/ppa \
```
Comment (fcyu) [inline Dockerfile:6]: I don't think these two are required.

File: Dockerfile [hunk 2]
```diff
@@ -1,41 +1,39 @@
...
+RUN apt-get -yqq install \
+    fontconfig \
     git \
-    python3.9 \
+    libglib2.0-0 \                                                                      
+    libsm6 \                                                                            
+    libxrender1 \                                                                       
+    libxext6 \
+    libfreetype6 \                                                                      
+    libgomp1 \
```
Comment (fcyu) [inline Dockerfile:17]: Could you tell me why do you install these packages for FragPipe? The original dockfile without those packages works fine.

File: Dockerfile [hunk 3]
```diff
@@ -1,41 +1,39 @@
...
+    python3.12 \
```
Comment (fcyu) [inline Dockerfile:18]: EasyPQP requires Python 3.9, which has been included in the original dockerfile.

File: Dockerfile [hunk 4]
```diff
@@ -1,41 +1,39 @@
...
+    libfreetype6 \                                                                      
+    libgomp1 \
+    python3.12 \                                                                        
     python3-pip \
     tar \
-    unzip \
-    wget \
-    openjdk-11-jdk \
-	vim \
-	dotnet-runtime-6.0
```
Comment (fcyu) [inline Dockerfile:21]: FragPipe requires Java 9+, so Java 11 is used here. Its O-Pair module requires dotnet 6.0.

File: Dockerfile [hunk 5]
```diff
@@ -1,41 +1,39 @@
...
-# set the working directory to the HOME of root
-WORKDIR /root
-
-# create directories
-RUN mkdir tmp
-	
+    && pip install lxml \
+    && pip install --force-reinstall numpy==1.25 \
+    && pip install --force-reinstall pandas==1.5.3 \
+    && pip install pyopenms
```
Comment (fcyu) [inline Dockerfile:29]: I don't think those three are necessary. When installing EasyPQP, they will be installed ad dependencies.

File: Dockerfile [hunk 6]
```diff
@@ -1,41 +1,39 @@
...
+# Install OpenJDK-8
+RUN apt-get update && \
+    apt-get install -y openjdk-18-jdk && \
```
Comment (fcyu) [inline Dockerfile:32]: The original dockfile has Java 11 installed. We never tested Java 18 thoroughly.

**Review by fcyu — CHANGES_REQUESTED** [review]
Most of the changes are not necessary or not appropriate. Could you please give more info about why making those changes?

Thanks,

Fengchao

File: Dockerfile [hunk 7]
```diff
@@ -1,41 +1,39 @@
...
+    apt-get clean;
 # download and install fragPipe
-RUN wget https://github.com/Nesvilab/FragPipe/releases/download/21.0/FragPipe-21.0.zip -P fragPipe-21.0
-RUN unzip fragPipe-21.0/FragPipe-21.0.zip -d fragPipe-21.0
-RUN chmod -R 770 fragPipe-21.0
-
+RUN wget https://github.com/Nesvilab/FragPipe/releases/download/21.1/FragPipe-21.1.zip -P /home/fragPipe-21.1
+RUN unzip /home/fragPipe-21.1/FragPipe-21.1.zip -d /home/fragPipe-21.1
```
Comment (fcyu) [inline Dockerfile:36]: You were looking at the file in the `master` branch, but you should use the `develop` branch.

**Review by fcyu — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (rolivella) [reply Dockerfile:6]: Ok, for me are required, but if it works without them, better to do not add them!

File: Dockerfile (hunk 2 above)
Comment (rolivella) [reply Dockerfile:17]: I don't exactly know. But the same for the previous comment, better to do not add them!

**Comment by rolivella** [comment]
In summary, I think that the only section that should be removed is, I think is not necessary: 

```
# set the working directory to the HOME of root
WORKDIR /root

# create directories
RUN mkdir tmp
```

I'll create the proper PR from dev branch for this two.

**Pull request closed without merging**